src/ui/ResponsiveComponents/ResponsiveFacetColumn.ts

```diff
@@ -94,7 +88,23 @@ export class ResponsiveFacetColumn implements IResponsiveComponent {
   }
 
   private needSmallMode(): boolean {
-    return this.coveoRoot.width() <= this.breakpoint;
+    if (this.searchInterface) {
+      switch (this.searchInterface.responsiveComponents.getResponsiveMode()) {
+        case 'small':
+          return true;
+        case 'auto':
+          return (
+            this.coveoRoot.width() <=
+            (Utils.isNullOrUndefined(this.breakpoint) ? this.searchInterface.responsiveComponents.getMediumScreenWidth() : this.breakpoint)
```

Comment:
Are the values returned by `this.searchInterface.responsiveComponents.getMediumScreenWidth()` and `new ResponsiveComponents().getMediumScreenWidth()` different? Or are they two ways to get the same thing?
If they are the same, we could cut the repetition by extracting the following into a separate method:

```
private get sanitizedBreakpoint() {
    return (Utils.isNullOrUndefined(this.breakpoint) ? new ResponsiveComponents().getMediumScreenWidth() : this.breakpoint;
}
```

Going a step further, why not sanitize in the constructor? I see we used to do that in the red part of the diff lines 58-64, so I'm guessing there was an issue with that?

Reply:
Another good method would be:

```
private isWidthSmallerThanBreakpoint() {
    this.coveoRoot.width() < this.sanitizedBreakpoint
}
```

I removed the equals, but if that precision is needed, then the name would become `widthSmallerThanOrEqualToBreakpoint`